### PR TITLE
Add notification for both recording and transcription activity on the floating bar

### PR DESCRIPTION
**Pull request by lpeixin**
Add notification bar for both recording and transcription activity to cover different cases of notification text.

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 1]
```diff
@@ -363,13 +376,46 @@ private void updateParticipantViews() {
...
+        } else {
+            if (isTranscriptionActive) {
+                setRecordingActivityNotification(getText(R.string.stop_recording_while_transcribing));
+            } else {
+                setRecordingActivityNotification(getText(R.string.stop_recording));
+            }
+        }
+    }
+
+    private void showTranscriptionActiveNotification() {
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:397]: I think we can combine these methods into one. They have the same logic to check `isRecordingActive` and `isTranscriptionActive` and display the text based on the two boolean values.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 2]
```diff
@@ -74,12 +74,14 @@
     private boolean micOn;
     private boolean isVideoOnHold = false;
     private boolean isRecordingActive = false;
+    private Boolean isTranscriptionActive = false;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:77]: We can just use the primitive `boolean`.

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 1 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:397]: Yea, originally I want to merge them into one. But the problem is for the case when one is true and the other is false. For this case, should we say "A has stopped. You are only B." or "B has started." if we merge them into one function? (basically in current logic, the function itself stands for the action coming from whom)

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java (hunk 2 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java]: addressed

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 3]
```diff
@@ -413,9 +459,8 @@ private void recycleTimer() {
 
     private void setRecordingActivityNotification(final CharSequence text) {
         runOnUiThread(() -> {
```
Comment (AmyL219) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:461]: You'd better rename the function, for example: setActivityNotification

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 1 above)
Comment (AmyL219) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:397]: Yeah, after rethinking, we could combine them into one method with input parameter, some similar logic to:

```
showNotification(fromRecordingTrigger:Bool) {
...
if (isRecordingActive) {
            if (isTranscriptionActive) {
               ...
            } else {
                fromRecordingTrigger ? setRecordingActivityNotification(getText(R.string.start_recording)) : setRecordingActivityNotification(getText(R.string.stop_transcribing_while_recording));
.....
```

Comment (waynemo) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:397]: Perhaps something like this:    

    private enum BannerType {
        RECORDING,
        TRANSCRIPTION
    }

    private void showBanner(BannerType bannerType) {
        if (callingContext.isRecordingActive && callingContext.isTranscriptionActive) {
            setBannerText(getText(R.string.start_recording_and_transcribing))
            return;
        }

        if (bannerType == BannerType.RECORDING) {
            if (callingContext.isRecordingActive) {
                setBannerText(getText(R.string.start_recording))
            } else {
                callingContext.isTranscriptionActive
                        ? setBannerText(getText(R.string.stop_recording_while_transcribing))
                        : setBannerText(getText(R.string.stop_recording));
            }
        } else if (bannerType == BannerType.TRANSCRIPTION) {
            // Similar logic for transcription
        }
    }

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 4]
```diff
@@ -65,7 +65,8 @@
     private ImageButton audioImageButton;
     private LinearLayout callHangupOverlay;
     private View infoHeaderView;
-    private View recordingActivityHeader;
+    private View floatingHeaderArea;
+    private TextView notificationActivityText;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:69]: Let's call this `noticeBannerText`

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 5]
```diff
@@ -65,7 +65,8 @@
     private ImageButton audioImageButton;
     private LinearLayout callHangupOverlay;
     private View infoHeaderView;
-    private View recordingActivityHeader;
+    private View floatingHeaderArea;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:68]: Let's call this `noticeBannerView`.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 3 above)
Comment (waynemo) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:461]: Let's call this `setNoticeBannerText`

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 6]
```diff
@@ -591,7 +636,8 @@ private void setupScreenLayout() {
         callHangupCancelButton.setOnClickListener(l -> closeHangupDialog());
 
         infoHeaderView = findViewById(R.id.info_header);
-        recordingActivityHeader = findViewById(R.id.recordingActivityHeader);
+        floatingHeaderArea = findViewById(R.id.recordingActivityHeader);
+        notificationActivityText = findViewById(R.id.recordingNotification);
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:640]: Let's rename the ids in the xml:

- recordingActivityHeader->noticeBanner
- recordingNotification->noticeBannerText
- closeRecordingNotification->closeNoticeBannerButton

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 7]
```diff
@@ -363,13 +376,46 @@ private void updateParticipantViews() {
...
+                setRecordingActivityNotification(getText(R.string.stop_transcribing_while_recording));
+            } else {
+                setRecordingActivityNotification(getText(R.string.stop_transcribing));
+            }
         }
-        recordingActivityHeader.setVisibility(View.VISIBLE);
-        recordingActivityHeader.bringToFront();
+    }
+
+    private void setFloatingHeaderVisible() {
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:416]: Let's rename to `showNoticeBanner`

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 1 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: Yea, addressed. Originally I thought the code in this way doesn't save lines of code. but yea, it saves number of functions and merged some cases.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 3 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:451]: ok, addressed.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 4 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: updated.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 5 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: addressed.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 6 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: oh, I missed.. now address and updated them.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 7 above)
Comment (lpx1989) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: addressed.

**Review by iaulakh — APPROVED** [review]

**Pull request merged**